lib/twilio/call.ts

```diff
@@ -851,6 +862,45 @@ class Call extends EventEmitter {
     }
   }
 
+  /**
+   * Send a message to Twilio.
+   *
+   * @example
+   * ```ts
+   * call.sendMessage(Call.MessageType.UserDefinedMessage, {
+   *   ahoy: 'world!,
+   * });
+   * ```
+   *
+   * @param messageType - The type of the message to send to Twilio.
+   * @param content - The content of the message to send to Twilio.
+   */
+  sendMessage(messageType: Call.MessageType, content: any) {
+    if (typeof messageType !== 'string') {
+      throw new InvalidArgumentError(
+        'Message type must be an enumeration value of `Call.MessageType` or ' +
+        'a string.',
+      );
+    }
+
+    if (this._pstream === null) {
+      throw new InvalidStateError(
+        'Could not send CallMessage; Signaling channel is disconnected',
+      );
+    }
+
+    const callSid = this.parameters.CallSid;
+    if (typeof callSid !== 'string') {
+      throw new InvalidStateError(
+        'Could not send CallMessage; Call has no CallSid',
+      );
+    }
```

Comment:
I think it's better to check call status so you don't need these 2 checks. You will also probably cover more scenario. Something like
```ts
if (this._status !== Call.State.Open) {
  throw new InvalidStateError(
      `Could not send message. Call status is ${this._status}`
  );
}
``

Reply:
Sounds good, will do.

Reply:
Do you think it's still worth type-guarding `this.parameters.CallSid`?

Reply:
Do you mean keep the `typeof callSid !== 'string'` check? I don't think we need it. Check the status that it is open should cover that.

Reply:
@mhuynh5757 I just learned base on this [discussion](https://twilio.slack.com/archives/C02U2QQCMME/p1663964471252479) that client should be able to receive events outside of "open" status, such as ringing. This means, we should only check here if there's a websocket connection or not.

Reply:
What about a CallSid? This is sending not receiving right? Seems kind of weird to want to send a message during ringing.

Reply:
It's a valid use case. The customer can use it to send status events to their backend or other clients.

Reply:
Ok, sounds solid then.

Reply:
Do we want to add the CallSid check back in?

Reply:
No, we don't want to check for callsid. I don't think callsid is generated until the call is established no?

Reply:
Yes, but we always send the `CallSid` with every `message` package over VSP.